### ISSUE TITLE
Adding organization/team information for users

### ISSUE THREAD

**Issue by ft-cama** [jupyterhub/zero-to-jupyterhub-k8s#2633]
### Proposed change
Adding organization/team information dynamically for each user.

### Who would use this feature?
Anyone who's shipping Jupyterhub for multiple teams/organizations, and wants to customize the user's Jupyterhub server instance based on what organization(s)/team(s) that user is part of.

### Why is this feature needed?
For example: Jupyterhub is being used for a computer science class and the students are working in separate teams, then it would be very beneficial to have that team information, so that a team-specific volume is mounted whenever a student starts his/her Jupyterhub server.

It is possible already now to make team-based volumes, given that you hardcode the usernames and corresponding team(s) into your config.yaml. But hardcoding usernames and teams is more error-prone compared to just having the authenticator pass on the username and corresponding team information. Furthermore, if a user is added to a team or removed from a team, you would have to manually update the user/teams list in config.yaml, and because it is manual, this process is slow and error-prone.

A further enhancement could be to have the possibility in config.yaml to specify that Jupyterhub should handle an NFS server and create team-directories automatically and mount a user's team-specific directories when the user logs in, which requires that a user's team(s) information is available.

In my own config file I have a custom KubeSpawner class doing something in that direction, but of course the persistent volume and persistent volume claim for each org has been created manually at this point. Each persistent volume points to a directory on an NFS server named after that org.

```
class CustomSpawner(KubeSpawner):
    @gen.coroutine
    def start(self):
        auth_state = yield self.user.get_auth_state()
        for org in auth_state['user_orgs']:
            self.volumes += [{'name' : 'nfs_pv_'+org, 'persistentVolumeClaim' : {'claimName' : 'nfs_pvc_'+org}}]
            self.volume_mounts += [{'mountPath' : '/home/'+org, 'name' : 'nfs_pv_'+org}]
        return (yield super().start())
```

### (Optional): Suggest a solution
I have for the [GithubOAuthenticator](https://github.com/jupyterhub/oauthenticator/blob/8c1cf0c616d5e19d1f93238673f17db2daf57fa0/oauthenticator/github.py) proposed a solution in this [issue](https://github.com/jupyterhub/oauthenticator/issues/492) for getting organization/teams information for each user. 

This is just one step in the right direction. This needs to be done for all authenticators that support something like teams/organizations. 

If teams/organizations information becomes available, then this opens up for very nice features, e.g. automatically mounting team-specific volumes.

**Comment by consideRatio** [comment]
Thanks for a nice writeup @ft-cama! I'll go ahead and close this issue in this repo as it isn't suitable to be tracked here, but the you have in jupyterhub/oauthenticator is very concrete and relevant to the repo - which makes it something easy to work towards implementing, in that specific repo.

If you want to raise the more general discussion, I suggest going for a discourse.jupyter.org post and searching for "group" or "groups" in jupyterhub/jupyterhub issues/prs as I recall there is some work related to that - which could be of relevance to the discussion.

Btw, this post I wrote on discourse is related to doing what you do. In it, you have an example where @gen.coroutine has been replaced with an async function which is the now supported and modern way to use going onwards: https://discourse.jupyter.org/t/tailoring-spawn-options-and-server-configuration-to-certain-users/8449
